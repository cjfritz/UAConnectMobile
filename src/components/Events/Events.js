import React, { PureComponent } from 'react';
import { WebView } from 'react-native';
import {
  Container, Button, Icon, Header, Left, Body, Right, Title,
} from 'native-base';
import styles from './Events.style';

class Events extends PureComponent {
  state = {
    canGoBack: false,
  };

  webview = null;

  goBack = () => {
    this.webview.goBack();
  };

  render() {
    const { canGoBack } = this.state;
    return (
      <Container>
        <Header style={ styles.header }>
          <Left style={ styles.headerSection }>
            <Button
              transparent
              disabled={ !canGoBack }
              onPress={ this.goBack }
            >
              <Icon name='arrow-back' />
            </Button>
          </Left>
          <Body style={ styles.headerSection }>
            <Title style={ styles.title }>Events</Title>
          </Body>
          <Right style={ styles.headerSection } />
        </Header>
        <WebView
          source={ { uri: 'https://calendars.uark.edu/EventList.aspx?fromdate=12%2f2%2f2018&todate=12%2f2%2f2018&display=Day&view=DateTime' } }
          ref={ ref => (this.webview = ref) }
          onNavigationStateChange={ navState => {
            this.setState({ canGoBack: navState.canGoBack });
          } }
        />
      </Container>
    );
  }
}

export default Events;

