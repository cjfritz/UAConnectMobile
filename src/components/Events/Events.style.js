import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  backButton: {
    marginTop: '6%',
  },
  headerView: {
    backgroundColor: 'darkred',
    flexDirection: 'row',
    height: '12%',
    width: '100%',
  },
});
