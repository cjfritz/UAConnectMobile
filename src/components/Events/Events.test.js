import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Events from './Events';

describe('<Events />', () => {
  it('renders with default props', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<Events />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});
