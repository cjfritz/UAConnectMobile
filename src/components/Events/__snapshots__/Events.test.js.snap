// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Events /> renders with default props 1`] = `
<Styled(Container)>
  <Component
    style={
      Object {
        "backgroundColor": "darkred",
        "flexDirection": "row",
        "height": "12%",
        "width": "100%",
      }
    }
  >
    <Styled(Button)
      disabled={true}
      onPress={[Function]}
      style={
        Object {
          "marginTop": "6%",
        }
      }
      transparent={true}
    >
      <Styled(Icon)
        name="arrow-back"
      />
    </Styled(Button)>
  </Component>
  <WebView
    onNavigationStateChange={[Function]}
    originWhitelist={
      Array [
        "http://*",
        "https://*",
      ]
    }
    source={
      Object {
        "uri": "https://registrar.uark.edu/academic-dates/academic-semester-calendar/index.php",
      }
    }
  />
</Styled(Container)>
`;
